src/PaymentGateways/PayPalCommerce/AjaxRequestHandler.php

```diff
@@ -104,21 +113,133 @@ public function onGetPartnerUrlAjaxRequestHandler() {
 	 * @since 2.8.0
 	 */
 	public function removePayPalAccount() {
-		$this->sendErrorOnAjaxRequestIfUserDoesNotHasPermission();
+		$this->validateAdminRequest();
 
 		give( MerchantDetail::class )->delete();
 
 		wp_send_json_success();
 	}
 
 	/**
-	 * Send error if user does not has capability to manage GiveWP settings.
+	 * Create order.
+	 *
+	 * @todo: handle payment create error on frontend.
 	 *
 	 * @since 2.8.0
 	 */
-	private function sendErrorOnAjaxRequestIfUserDoesNotHasPermission() {
+	public function createOrder() {
+		$this->validateFrontendRequest();
+
+		/* @var PayPalHttpClient $client */
+		$client = give( PayPalClient::class )->getHttpClient();
+		/* @var MerchantDetail $merchant */
+		$merchant = give( MerchantDetail::class );
+
+		$postData = give_clean( $_POST );
+		$formId   = absint( $postData['give-form-id'] );
+
+		$request = new OrdersCreateRequest();
+		$request->payPalPartnerAttributionId( PartnerDetails::$attributionId );
+		$request->body = [
+			'intent'              => 'CAPTURE',
+			'purchase_units'      => [
+				[
+					'reference_id'        => get_post_field( 'post_name', $formId ),
+					'description'         => '',
+					'amount'              => [
+						'value'         => give_maybe_sanitize_amount( $postData['give-amount'] ),
+						'currency_code' => give_get_currency( $_POST['give-form-id'] ),
+					],
+					'payee'               => [
+						'email_address' => $merchant->merchantId,
+						'merchant_id'   => $merchant->merchantIdInPayPal,
+					],
+					'payer'               => [
+						'given_name'    => $postData['give_first'],
+						'surname'       => $postData['give_last'],
+						'email_address' => $postData['give_email'],
+					],
+					'payment_instruction' => [
+						'disbursement_mode' => 'INSTANT',
+					],
+				],
+			],
+			'application_context' => [
+				'shipping_preference' => 'NO_SHIPPING',
+				'user_action'         => 'PAY_NOW',
+			],
+		];
+
+		try {
+			$response = $client->execute( $request );
+
+			wp_send_json_success(
+				[
+					'id' => $response->result->id,
+				]
+			);
+		} catch ( \Exception $ex ) {
+			wp_send_json_error(
+				[
+					'errorMsg' => $ex->getMessage(),
+				]
+			);
+		}
+	}
+
+	/**
+	 * Approve order.
+	 *
+	 * @todo: handle payment capture error on frontend.
+	 *
+	 * @since 2.8.0
+	 */
+	public function approveOrder() {
+		$this->validateFrontendRequest();
+
+		/* @var PayPalHttpClient $client */
+		$client  = give( PayPalClient::class )->getHttpClient();
```

Comment:
Once you're using DI in the class you won't have to grab this from the container multiple times.

Reply:
You already updated code in your webhook pr. I will fix that after merging your pr.

Reply:
fixed: https://github.com/impress-org/givewp/compare/f62f6b626cee...ded2242363cc